Source/cmake/OptionsGTK.cmake

```
@@ -28,6 +28,10 @@ find_package(ATSPI 2.5.3)
 include(GStreamerDefinitions)
 include(FindGLibCompileResources)
 
+find_package(Execinfo)
+SET_AND_EXPOSE_TO_BUILD(HAVE_BACKTRACE_FUNCTION ${Execinfo_FOUND})
+
+
```

Please be attentive to the whitespace expectations. Having two blank lines here is weird; the surrounding code does not do that:

```suggestion
```

---

@mcatanzaro That's only true for glibc. There _is_ a `libexecinfo` which some Musl systems ship, and FreeBSD has such a library, too. So no, it is not “weird” to check for a library that we may need to be able to use `backtrace()`. And we want the imported target to refer to the library when available.

---

Hm, OK, then using an imported target really does make sense, and my suggestion to put this in OptionsCommon.cmake doesn't work.

I also see that I was wrong about the imported target not doing anything. It actually does link in the libexecinfo library if `NOT HAVE_BACKTRACE AND HAVE_LIBEXECINFO`. That actually makes sense, but it's confusing and needs to be documented.

We need some more changes:

 * We need to make sure it's not an automagic dependency. The build should fail if libexecinfo is not found. You can add a new build option to allow building without libexecinfo if desired.
 * The imported target should itself provide the required definitions, and not define things globally.
 * The find module should document how it works, including all variables it defines and the target itself. Undefine anything that doesn't make sense to document.

---

Provided that `backtrace()` and `backtrace_symbols()` may be provided by the libc _or_ alternatively by libexecinfo, in this particular case, and as an exception, I lean towards handling this automagically. The same way that linking librt is handled automagically.

Actually using `target_compile_definitions(Execinfo::Execinfo INTERFACE ...)` is a nice idea, and it does sound better than adding them where needed in different CMake files. It feels a bit awkward to have a `PAS_` prefixed definition in the `Execinfo::Execinfo` target, but I can live with that.

I agree that documenting the find-module would be good (@AlexsanderDamaceno: you can look at other modules to see how to document them).